src/transformStateWithClones.js

```diff
@@ -8,6 +8,41 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = { ...state };
+  const arrResult = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+
+        for (const key in extraData) {
+          result[key] = extraData[key];
+        };
+        break;
+
+      case 'removeProperties':
+
+        for (const toRemove of keysToRemove) {
+          for (const key in result) {
+            if (toRemove === key) {
+              delete result[key];
+            };
+          };
```

Comment:
redundant nested loop, you shouldn't check keys, by the task - this keys 100% is keys of state